totp/services/config/config.c

```diff
@@ -53,6 +54,22 @@ FlipperFormat* totp_open_config_file(Storage* storage) {
             totp_close_config_file(fff_data_file);
             return NULL;
         }
+    } else if(storage_common_stat(storage, CONFIG_FILE_PATH_PREVIOUS, NULL) == FSE_OK) {
+        FURI_LOG_D(LOGGING_TAG, "Old config file %s found", CONFIG_FILE_PATH_PREVIOUS);
+        if(storage_common_stat(storage, CONFIG_FILE_DIRECTORY_PATH, NULL) == FSE_NOT_EXIST) {
+            FURI_LOG_D(
+                LOGGING_TAG,
+                "Directory %s doesn't exist. Will create new.",
+                CONFIG_FILE_DIRECTORY_PATH);
+            if(!storage_simply_mkdir(storage, CONFIG_FILE_DIRECTORY_PATH)) {
+                FURI_LOG_E(LOGGING_TAG, "Error creating directory %s", CONFIG_FILE_DIRECTORY_PATH);
+                return NULL;
+            }
+        }
+        storage_common_copy(storage, CONFIG_FILE_PATH_PREVIOUS, CONFIG_FILE_PATH);
```

Comment:
Also I think it worth it to check if "new config file" exists before even trying to copy old one. Because if "new file" already exists, we definitely should not try to overwrite it.

Reply:
New config file shouldn't be existing here, otherwise it would enter in the first if

Reply:
Yeah, agree, you are right